lib/Dialect/Moore/MooreOps.cpp

```diff
@@ -265,6 +265,19 @@ VariableOp::handlePromotionComplete(const MemorySlot &slot, Value defaultValue,
   return std::nullopt;
 }
 
+LogicalResult VariableOp::canonicalize(VariableOp op,
+                                       ::mlir::PatternRewriter &rewriter) {
+  for (auto *user : op->getUsers())
+    if (isa<ContinuousAssignOp>(user) &&
+        (user->getOperand(0) == op.getResult())) {
+      auto assignVarOp = rewriter.create<AssignedVarOp>(
+          op->getLoc(), op.getType(), op.getName(), user->getOperand(1));
+      rewriter.replaceOp(op, assignVarOp);
+      return success();
+    }
+  return failure();
+}
```

Comment:
I think you must ensure that  there is no multiple `ContinuousAssignOp`.

Reply:
Slang has guaranteed that the variable has only one continuous assignment. For example:
```
int x;
assign x = 32;
assign x = 64;
```
```
source.sv:4:9: error: cannot have multiple continuous assignments to variable 'x'
 assign x = 64;
        ^
source.sv:3:9: note: also assigned here
 assign x = 32;
 ```

Reply:
I think we shouldn't rely on implicit assumption of slang. We should at least check that in our verifier.

Reply:
Ok, I'll add this.

Reply:
I understood why we need to check whether there are multiple continuous assignments for the same variable. Because the `--canonicalize` operates on the IR, not from AST to IR, right?

Reply:
Yes, I think we should able to use Moore dialect not only for slang.